homeassistant/components/vodafone_station/coordinator.py

```diff
@@ -95,15 +95,19 @@ async def _async_update_data(self) -> UpdateCoordinatorDataType:
         """Update router data."""
         _LOGGER.debug("Polling Vodafone Station host: %s", self._host)
         try:
-            logged = await self.api.login()
-        except exceptions.CannotConnect as err:
-            _LOGGER.warning("Connection error for %s", self._host)
-            raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
-        except exceptions.CannotAuthenticate as err:
-            raise ConfigEntryAuthFailed from err
-
-        if not logged:
-            raise ConfigEntryAuthFailed
+            try:
+                await self.api.login()
+            except exceptions.CannotAuthenticate as err:
+                raise ConfigEntryAuthFailed from err
+            except (
+                exceptions.CannotConnect,
+                exceptions.AlreadyLogged,
+                exceptions.GenericLoginError,
+            ) as err:
+                raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
+        except (ConfigEntryAuthFailed, UpdateFailed):
```

Comment:
I would probably have gone with plain Exception here, but should be good none the less.